### PR TITLE
[15.0][ADD] base_multi_branch_company

### PR DESCRIPTION
**Pull request by Saran440**
This module allows you to add multiple branches within each company

Selection_004

You can add multi branch in company
Selection_005

You can see module implement from this module
- account_multi_branch - https://github.com/OCA/account-invoicing/pull/1490

### CODE REVIEW

File: base_multi_branch/views/res_branch_view.xml [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+                            <field name="company_id" />
+                            <field name="active" invisible="1" />
+                        </group>
+                    </group>
+                </sheet>
+            </form>
+        </field>
+    </record>
+    <record id="action_res_branch" model="ir.actions.act_window">
+        <field name="name">Branch</field>
```
Comment (rousseldenis) [inline base_multi_branch/views/res_branch_view.xml]:
```suggestion
        <field name="name">Branches</field>
```

**Review by rousseldenis — APPROVED** [review]
Code review. Minor comment

File: base_multi_branch/views/res_branch_view.xml (hunk 1 above)
Comment (Saran440) [reply base_multi_branch/views/res_branch_view.xml]: Fixed. Thank you for your check :)

**Review by ps-tubtim — APPROVED** [review]
LGTM

**Comment by Saran440** [comment]
@OCA/tools-maintainers Hi, someone help review and merge please.

**Comment by simahawk** [comment]
Isn't `base_multi_branch_company` a more appropriated name?

**Comment by Saran440** [comment]
@simahawk Ok, it's make sense for name `base_multi_branch_company`. Thank you for reviewed.

**Comment by thomaspaulb** [comment]
Could you explain more about the use case of this module? Eg when would you go multi-company, when would you go for operating unit, and when would you use branches?

**Comment by thomaspaulb** [comment]
@Saran440 There's also a pre-commit failure.

**Comment by Saran440** [comment]
@thomaspaulb This module used for accounting report.

For me, operating unit has rule for visible document. some user can't see OU but need select branch for report.

**Review by thomaspaulb — APPROVED** [review]

**Comment by thomaspaulb** [comment]
@Saran440 please still address pre-commit issue

**Pull request merged**